app/uk/gov/hmrc/agentservicesaccount/controllers/desiDetails/SelectDetailsController.scala

```diff
@@ -0,0 +1,85 @@
+ */
+
+package uk.gov.hmrc.agentservicesaccount.controllers.desiDetails
+
+import play.api.Logging
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import uk.gov.hmrc.agentservicesaccount.actions.Actions
+import uk.gov.hmrc.agentservicesaccount.config.AppConfig
+import uk.gov.hmrc.agentservicesaccount.controllers.desiDetails.util.NextPageSelector.getNextPage
+import uk.gov.hmrc.agentservicesaccount.controllers.{CURRENT_SELECTED_CHANGES, PREVIOUS_SELECTED_CHANGES, ToFuture}
+import uk.gov.hmrc.agentservicesaccount.forms.SelectChangesForm
+import uk.gov.hmrc.agentservicesaccount.models.SelectChanges
+import uk.gov.hmrc.agentservicesaccount.repository.UpdateContactDetailsJourneyRepository
+import uk.gov.hmrc.agentservicesaccount.services.SessionCacheService
+import uk.gov.hmrc.agentservicesaccount.views.html.pages.desi_details.select_changes
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SelectDetailsController @Inject()(actions: Actions,
+                                        val updateContactDetailsJourneyRepository: UpdateContactDetailsJourneyRepository,
+                                        sessionCache: SessionCacheService,
+                                        select_changes_view: select_changes
+                                       )(implicit appConfig: AppConfig,
+                                         cc: MessagesControllerComponents,
+                                         ec: ExecutionContext
+                                       ) extends FrontendController(cc)
+  with UpdateContactDetailsJourneySupport with I18nSupport with Logging{
+
+  def ifFeatureEnabled(action: => Future[Result]): Future[Result] = {
+    if (appConfig.enableChangeContactDetails) action else Future.successful(NotFound)
+  }
+
+  def showPage: Action[AnyContent] = actions.authActionCheckSuspend.async { implicit request =>
+    actions.ifFeatureEnabled(appConfig.enableNonHmrcSupervisoryBody) {
+      Ok(select_changes_view(SelectChangesForm.form)).toFuture
+    }
+  }
+
+  def onSubmit: Action[AnyContent] = actions.authActionCheckSuspend.async { implicit request =>
+    actions.ifFeatureEnabled(appConfig.enableNonHmrcSupervisoryBody) {
+      SelectChangesForm.form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => {
+            Ok(select_changes_view(formWithErrors)).toFuture
+          },
+          (selectedChanges: SelectChanges) => {
+            sessionCache.put(CURRENT_SELECTED_CHANGES, selectedChanges.pagesSelected).flatMap {
+              _ => {
+                removeUnchecked(selectedChanges)
+                getNextPage(sessionCache)
+              }
+            }
+          }
+        )
+    }
+  }
+
+
+  def removeUnchecked(selectedChanges: SelectChanges)(implicit request: Request[_], ec: ExecutionContext): Unit = {
+    //ToDo: Remove session data for unchecked
```

Comment:
Is this TODO needed?

Reply:
Same as above, this needs doing but I think is beyond this ticket